operator/pkg/manifest/installer.go

```diff
@@ -993,3 +1000,44 @@ func logAndPrint(v ...interface{}) {
 	scope.Infof(s)
 	fmt.Println(s)
 }
+
+//extractClusterID from: values.global.clusterID -> uuid for 'kube-system`, return error if not found
+func extractClusterID(client kubernetes.Interface, iop *v1alpha1.IstioOperatorSpec) (string, error) {
+	iopvalString := util.ToYAML(iop.Values)
+	values := &iopv1alpha1.Values{}
+	if err := util.UnmarshalValuesWithJSONPB(iopvalString, values, true); err != nil {
+		return "", fmt.Errorf("failed to parse global values in IOP: %v", err)
+	}
+	clusterID := values.Global.ClusterID
+	if clusterID == "" {
+		n, err := client.CoreV1().Namespaces().Get(context2.TODO(), "kube-system", metav1.GetOptions{})
+		if err != nil {
+			return "", fmt.Errorf("failed to get uid for namespace %s to use as clusterID", "kube-system")
+		}
+		clusterID = string(n.UID)
+	}
+	return clusterID, nil
+}
+
+func plugInExternalCertOrGenerateSelfSignedCerts(restConfig *rest.Config, certDir string, iop *v1alpha1.IstioOperatorSpec, dryRun bool) error {
+	//try to plugIn external certs
+	client, err := kubernetes.NewForConfig(restConfig)
```

Comment:
client is only used when certDir != "", can you just do the following in the beginning:
```
if certDir == "" {
  return nil
}
```